packages/auth0_api_python/tests/test_api_client.py

```diff
@@ -388,3 +404,1153 @@ async def test_verify_access_token_fail_no_audience_config():
+        domain="auth0.local",
+        user_id="test_user",
+        audience="my-audience",
+    )
+
+    api_client = ApiClient(
+        ApiClientOptions(domain="auth0.local", audience="my-audience")
+    )
+
+    # Test Bearer scheme
+    result = await api_client.verify_request(
+        headers={"authorization": f"Bearer {token}"},
+        http_method="GET",
+        http_url="https://api.example.com/resource"
+    )
+
+    assert "sub" in result
+    assert result["aud"] == "my-audience"
+    assert result["iss"] == "https://auth0.local/"
+
+@pytest.mark.asyncio
+async def test_verify_request_dpop_scheme_success(httpx_mock: HTTPXMock):
+    """
+    Test successful DPoP token verification through verify_request.
+    """
+    # Mock OIDC discovery
+    httpx_mock.add_response(
+        method="GET",
+        url="https://auth0.local/.well-known/openid-configuration",
+        json={
+            "jwks_uri": "https://auth0.local/.well-known/jwks.json",
+            "issuer": "https://auth0.local/",
+        },
+    )
+
+    # Mock JWKS endpoint
+    httpx_mock.add_response(
+        method="GET",
+        url="https://auth0.local/.well-known/jwks.json",
+        json={"keys": [PUBLIC_RSA_JWK]},
+    )
+
+    # Generate DPoP bound token and proof
+    access_token = await generate_token_with_cnf(
+        domain="auth0.local",
+        user_id="test_user",
+        audience="my-audience",
+    )
+    dpop_proof = await generate_dpop_proof(
+        access_token=access_token,
+        http_method="GET",
+        http_url="https://api.example.com/resource"
+    )
+
+    api_client = ApiClient(
+        ApiClientOptions(domain="auth0.local", audience="my-audience")
+    )
+
+    # Test DPoP scheme
+    result = await api_client.verify_request(
+        headers={"authorization": f"DPoP {access_token}", "dpop": dpop_proof},
+        http_method="GET",
+        http_url="https://api.example.com/resource"
+    )
+
+    assert "sub" in result
+    assert result["aud"] == "my-audience"
+    assert result["iss"] == "https://auth0.local/"
+
+
+# --- Configuration & Error Handling Tests ---
+
+@pytest.mark.asyncio
+async def test_verify_request_fail_dpop_required_mode():
+    """
+    Test that Bearer tokens are rejected when DPoP is required.
+    """
+    # Generate a valid Bearer token
+    token = await generate_token(
+        domain="auth0.local",
+        user_id="test_user",
+        audience="my-audience",
+    )
+
+    api_client = ApiClient(
+        ApiClientOptions(
+            domain="auth0.local",
+            audience="my-audience",
+            dpop_required=True  # Require DPoP
+        )
+    )
+
+    with pytest.raises(InvalidAuthSchemeError) as err:
+        await api_client.verify_request(
+            headers={"authorization": f"Bearer {token}"},
+            http_method="GET",
+            http_url="https://api.example.com/resource"
+        )
+
+    assert "dpop" in str(err.value).lower() or "bearer" in str(err.value).lower()
```

Comment:
It's not clear the assertion is verifying that the Bearer token is being rejected.